stdlib/proposals/websubhub/subscription_autoverification.md

```diff
@@ -0,0 +1,124 @@
+# Auto-verify subscriptions in Ballerina WebSubHub
+
+- Authors
+  - Ayesh Almeida
+- Reviewed by
+  - Shafreen Anfar, Danesh Kuruppu
+- Created date
+  - 2025-02-13
+- Issue
+  - [1336](https://github.com/ballerina-platform/ballerina-spec/issues/1336)
+- State
+  - Submitted
+
+## Summary
+
+This proposal introduces an enhancement to the Ballerina `websubhub` library that enables the developer to disable the WebSub 
+subscription verification process for well known or authenticated subscriptions. This proposal introduces a service-level 
+configuration to enable automatic subscription verification and an optional parameter for the `onSubscription` and 
+`onUnsubscription` methods in `websubhub:Service` to mark specific subscriptions as auto-verified. The identification of 
+properly authenticated subscriptions remains the responsibility of the developer implementing the WebSub hub.
+
+## Goals
+
+- Provide the WebSub hub implementer more control over how subscriptions are verified/authenticated.
+
+## Non-Goals
+
+- This proposal will not change the current behavior of the WebSub hub. It will only introduce an alternate mechanism which the developer can use to disable the current subscription verification process if needed.
+
+## Motivation
+
+The subscription intent verification step exists to verify and authenticate the subscription request sent to the hub. However, 
+in some scenarios, a developer may have well-known (and authenticated) subscribers who do not require this additional 
+verification step. This proposal aims to allow developers more control over the subscription verification process by 
+introducing a configuration in the Ballerina `websubhub` library. This will enable developers to skip the standard subscription 
+intent verification process based on the requirement.
+
+## Description
+
+This proposal introduces two levels of validation to enable subscription auto-verification in Ballerina WebSubHub:
+
+1. **Service-level configuration** :
+    - Add a new configuration parameter in the `@websubhub:ServiceConfig` annotation that allows enabling or disabling automatic subscription verification at the hub level.
+
+2. **Introduce an optional parameter for `onSubscription` and `onUnsubscription` methods of `websubhub:Service`** :
+    - Add a new client object that allows dynamically marking a subscription or unsubscription for auto-verification.
+
+### Service-level configuration
+
+An optional boolean configuration parameter will be added to the `@websubhub:ServiceConfig` annotation:
```

Comment:
A field with a default value of false (rather than an optional field)? Should probably also say annotation field rather than parameter.

Reply:
Fixed with [76f6433d9f64f80dccfcbdbbaa88f82f9c66c373](https://github.com/ballerina-platform/ballerina-spec/pull/1335/commits/76f6433d9f64f80dccfcbdbbaa88f82f9c66c373)